File: fixtures/objc/commoncrypto/sha256/main.m
#import <Foundation/Foundation.h>
#import <CommonCrypto/CommonCrypto.h>

int main() {
    @autoreleasepool {
        NSData *message = [@"Hello, World!" dataUsingEncoding:NSUTF8StringEncoding];
        uint8_t digest[CC_SHA256_DIGEST_LENGTH];
        
        CC_SHA256(message.bytes, (CC_LONG)message.length, digest);
    }
    return 0;
}

File: fixtures/swift/commoncrypto/sha256/main.swift
import Foundation
import CommonCrypto

let message = "Hello, World!".data(using: .utf8)!
var digest = [UInt8](repeating: 0, count: Int(CC_SHA256_DIGEST_LENGTH))

message.withUnsafeBytes { bytes in
    CC_SHA256(bytes.baseAddress, CC_LONG(message.count), &digest)
}
